README.md

```diff
@@ -8,223 +8,223 @@ extensions:
+
+> We will publish to DefinitelyTyped soon and then you can directly use `typings` to get access to the definitions.
+
+**If you are using any dependency loader** such as [RequireJS](http://requirejs.org/) or [SystemJS](https://github.com/systemjs/systemjs) or module bundler such as [browserify](http://browserify.org/), [webpack](https://webpack.github.io/), you can use TypeScript `import` syntax to import specific modules. For example, one of the following:
+
+```typescript
+import * as OfficeHelpers from '@microsoft/office-js-helpers';
+
+import {Authenticator, DefaultEndpoints} from '@microsoft/office-js-helpers';
+
+import {Authenticator, Storage} from '@microsoft/office-js-helpers';
+
+import {Authenticator} from '@microsoft/office-js-helpers';
+```
+
+## Helpers
+
+### Authentication
+
+The Authentication helper is built for standards compliant OAuth Implicit Flow. Out of the box it directly integrates with Microsoft, AzureAD, Google and Facebook authentication.
+
+> Microsoft integration uses the AzureAD AppModel v2 endpoints which uses Converged Authentication. It enables users to login using their Work, School or Personal accounts.
+
+> Note on MSAL. This helper isn't a replacement for MSAL. When MSAL for JavaScript is released publicly, the helper will use MSAL.
+
+#### For Office Add-ins
+You need to meet the following requirements before you are able to successfully to use the Authenticator inside of Office Add-ins.
+
+1. You need to use `https`. This is important as we are using OAuth Implicit Flow and it is critical to secure the communication over the wire.
+2. Add the location of the provider in the [AppDomains](https://docs.microsoft.com/office/dev/add-ins/reference/manifest/appdomains) section of your add-in's manifest, as shown in the following example:
+
+```xml
+    <AppDomain>https://login.windows.net</AppDomain>
+    <AppDomain>https://login.microsoftonline.com</AppDomain>
+```
+
+#### Setup the authenticator
+Inside of your [Office.initialize](https://docs.microsoft.com/office/dev/add-ins/develop/understanding-the-javascript-api-for-office#initialize-with-officeinitialize) function add the following check:
+
+```javascript
+if (OfficeHelpers.Authenticator.isAuthDialog()) return;
+```
+
+This to inform the Authenticator to automatically close the authentication dialog once the authentication is complete.
+
+> Note: This code needs to be run in the page that is redirected to from the provider. By default we assume the root url of your website. The code ensures that if an access_token, code or error was received inside of the dialog, then it will parse it and close the dialog automatically. Also as an additional step it ensures that the `state` sent to the provider is the same as what was returned, to prevent [Cross Site Request Forgery (CSRF)](http://www.twobotechnologies.com/blog/2014/02/importance-of-state-in-oauth2.html).

```

Comment:
```suggestion
> **Note**: This code needs to be run in the page that is redirected to from the provider. By default we assume the root URL of your website. The code ensures that if an access_token, code, or error was received inside of the dialog, then it will parse it and close the dialog automatically. Also as an additional step it ensures that the `state` sent to the provider is the same as what was returned, to prevent [Cross Site Request Forgery (CSRF)](http://www.twobotechnologies.com/blog/2014/02/importance-of-state-in-oauth2.html).
```